beacon-chain/sync/metrics.go

```diff
@@ -94,17 +92,11 @@ func (s *Service) updateMetrics() {
 	indices := s.aggregatorSubnetIndices(s.chain.CurrentSlot())
 	attTopic := p2p.GossipTypeMapping[reflect.TypeOf(&pb.Attestation{})]
 	attTopic += s.p2p.Encoding().ProtocolSuffix()
-	if featureconfig.Get().DisableDynamicCommitteeSubnets {
-		for i := uint64(0); i < params.BeaconNetworkConfig().AttestationSubnetCount; i++ {
-			formattedTopic := fmt.Sprintf(attTopic, digest, i)
-			topicPeerCount.WithLabelValues(formattedTopic).Set(float64(len(s.p2p.PubSub().ListPeers(formattedTopic))))
-		}
-	} else {
-		for _, committeeIdx := range indices {
-			formattedTopic := fmt.Sprintf(attTopic, digest, committeeIdx)
-			topicPeerCount.WithLabelValues(formattedTopic).Set(float64(len(s.p2p.PubSub().ListPeers(formattedTopic))))
-		}
+	for _, committeeIdx := range indices {
+		formattedTopic := fmt.Sprintf(attTopic, digest, committeeIdx)
+		topicPeerCount.WithLabelValues(formattedTopic).Set(float64(len(s.p2p.PubSub().ListPeers(formattedTopic))))
 	}
```

Comment:
I don't think this is a good idea. I think it should always be in the path of DisableDynamicCommitteeSubnets. 
If was on committee 5 and I have 10 peers on that subnet. Prometheus polls this metric, and the topic is updated with the value 10. 

Now, one epoch later, I am no longer on committee 5. When prometheus polls the metrics again, the value for committee subnet 5 is still 10 and it is not updated. This leads to stale and inaccurate results.